### PR TITLE
#4117: Added tip of the day functionality.

### PR DESCRIPTION
**Pull request by tchibatron**
I'm not sure that I need add another plugin to solve this issue. I guess my solition is also fine, but if I'm wrong can make it as a separate plugin.
Also I haven't found any ways to add config to General tab. To my mind it is not extensible. Because of this I've looked at other configs and found out, that all general custom configs are on PrefPageDatabaseGeneral, but may I'm also wrong :) I am open to any discussions.

Issue: Tip of the day popup #4117

### CODE REVIEW

**Comment by tchibatron** [comment]
I've created a separate plugin and changed the way of parsing tips file. Current parser supports only primitive html tags. 
And I also use an URL.openConnection to read the file with tips. There are some pros and cons. What do you think, is it OK?

**Comment by ruspl-afed** [comment]
Do you think it is possible to use just standard .properties file for tips definition?
```properties
tip_id1=Some <b>rich text</b> content
tip_id2=Another <u>cool</u> tip
```
This may simplify both implementation and localization.

**Comment by tchibatron** [comment]
If it is a property file, the support will be more difficult. Each time, when you need to add new tip, you will have to add a constat to java code, to get this tip as a property or think about the solution to dermine the count of tips and configure property id in runtime. I recon xml file is better in this case.

File: plugins/org.jkiss.dbeaver.core.application/plugin.xml [hunk 1]
```diff
@@ -50,6 +50,7 @@
             <property name="applicationCSSResources" value="platform:/plugin/org.eclipse.platform/images/"/>
             <property name="rendererFactoryUri" value="bundleclass://org.jkiss.dbeaver.core/org.jkiss.dbeaver.ui.e4.DBeaverRendererFactory"/>
             <property name="preferenceCustomization" value="plugin_customization.ini"/>
+            <property name="tipsFile" value="platform:/plugin/org.jkiss.dbeaver.ext.tipoftheday/tips.xml"/>
```
Comment (ruspl-afed) [inline plugins/org.jkiss.dbeaver.core.application/plugin.xml]: Please use `platform:/plugin/org.jkiss.dbeaver.ext.tipoftheday/$nl$/tips.xml` to support translation

File: plugins/org.jkiss.dbeaver.ext.tipoftheday/src/org/jkiss/dbeaver/ext/tipoftheday/TipOfTheDayInitializer.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        return result;
+    }
+
+
+    private InputStream getTipsFileStream() throws IOException {
+        String pathToTipsFile = Platform.getProduct().getProperty("tipsFile");
+
+        URL url = new URL(pathToTipsFile);
+        return url.openConnection().getInputStream();
```
Comment (ruspl-afed) [inline plugins/org.jkiss.dbeaver.ext.tipoftheday/src/org/jkiss/dbeaver/ext/tipoftheday/TipOfTheDayInitializer.java]: Please have a look at
`org.eclipse.core.runtime.FileLocator.find(URL)`

**Comment by ruspl-afed** [comment]
Constant in java code is not required to work with .properties, you can just use all the properties loaded with some `InputStream` like you do for .xml now.
Anyway, please do not forget to support localization of tips with `$nl$` segment in your URL

**Comment by serge-rider** [comment]
Going to merge but can't. DBeaverPreferences.java and DBeaverPreferencesInitializer.java were changed in every line (different line feeds I guess) so any change in them leads to a conflict :(

**Pull request merged**